base/show.jl

```diff
@@ -352,7 +352,7 @@ The following properties are in common use:
 
  - `:compact`: Boolean specifying that values should be printed more compactly, e.g.
    that numbers should be printed with fewer digits. This is set when printing array
-   elements. `:compact` output should not contain line breaks.
+   elements. `:compact` output for 2-argument [`show(io, x)`](@ref) should not contain line breaks.
```

Comment:
```suggestion
   elements. `:compact` output should not contain line breaks.
   This flag may be ignored for 3-argument [`show(io, mime, x)`](@ref).
```
I guess that this is what `show` for arrays and dictionaries does at present (apart from passing `:compact` on to the elements). It would also keep readers from asking themselves what "compact output with line breaks" may look like.

As expressed in the referenced Discourse thread, I would personally find a compact (human-readable) 3-argument `show` version quite useful. For this reason I suggest to add
```
Nevertheless, users are encouraged to take the `:compact` flag into account
when writing 3-argument `show` methods for the `"text/plain"` MIME type.
```